Bc review

bcaneco opened the pull request:
Hi guys, 

Submitting the pull request with changes arising from the review. These changes are addressing issues  #2, and #3, as well as other coding modifications and additions (e.g. check on EPSG, CRS, unit tests, etc)

It would be great if you go over the commits and accept/comment them as you see fit.

bcaneco committed:
changed order of function args

swapping positions of args to a more natural order

bcaneco committed:
added input check on EPSG

bcaneco committed:
added check on min timefilter

bcaneco committed:
minor stuff

bcaneco committed:
updated github link

bcaneco committed:
move bind_dist and bind_timediff outside bind_kmph condition

previously, options bind_dist and bind_timediff  were only evaluated if bind_kmph == TRUE. Was this as intended? If so, then this change needs to be reverted and appspec.json must be edited to reflect that dependency

bcaneco committed:
removed variable gap_mins

removed due to redundancy with timediff_hrs when bind_timediff == TRUE.

bcaneco committed:
added check on CRS

check on mandatory CRS setting in input data

bcaneco committed:
Parameter descriptions reworded and swapped order

Mainly optional rewording.  ESSENTIAL change to keep is that `speedcut`'s default MUST be NULL and not 0, otherwise the default is to filter out pretty much all observations!

bcaneco committed:
Update appspec.json

longer but perhaps clearer

bcaneco committed:
using "thinning" instead of "binning"

Binning usually refers to converting continuous values into discrete intervals, or ranking values according to a given discrete scale. So, using the term binning might be confusion (to me, at least, it was!). I think the term thinning provides a better description of what's being done

bcaneco committed:
wording edited

Just some changes in wording and to keep thinks consistent with appspec.json

bcaneco committed:
Update README.md

missed the previous commit

bcaneco committed:
item removed

unnecessary as these are specified as checkboxes in settings GUI - i.e. users can't leave these blank

bcaneco committed:
minor text edits

bcaneco committed:
code re-aranged, but same computation

bcaneco committed:
code re-arranged, same result

bcaneco committed:
minor re-write to enforce units in speedcut

bcaneco committed:
unit tests added

bcaneco committed:
typo

bcaneco commented:
Oh, something else I forgot: I wasn't able to replicate the errors that @callumjclarke  found with the units and the `mutate` function. Running sdk testing with `as.vector()` instances commented out went on as expected, with no errors thrown. 

For the moment I left the `as.vector()` statements active as they were, as I don't want to cause errors in subsequent apps. 

This is no big deal as there are bigger fish to fry atm, but reckon it would be neater to keep the units in all those columns where possible, for completeness and clarity when manipulating the data in the dependent apps.

bcaneco committed:
amendment to units of speedcut

need to remove this statement if units are being removed

bcaneco committed:
amendment for the removal of units in speedcut

lindesaysh commented on README.md:
```diff
@@ -2,23 +2,26 @@
+-   Create and append columns containing: 
+      - Universal Transverse Mercator (UTM) data
+      - An *index* column, consisting of the animal's ID and timestamp concatenated
+      - *speed*, *distance traveled* and/or *time since previous point* data
```
is it time since previous point or time to next point?

bcaneco replied:
Yep, you're absolutely right! Will amend this and commit again

lindesaysh commented on README.md:
```diff
@@ -45,46 +48,45 @@ Move2 location object
+`Bind additional timestamp columns` (logical): Determines whether to append specific timestamp data (*hour,* *minute,* *second,* *hourmin,* and *yearmonthday*) to the output. Default: `TRUE`.
 
-`Bind additional timestamp columns` (logical): Determines whether to append specific timestamp data (*hour,* *minute,* *second,* *hourmin,* and *yearmonthday*) to the output. 
+`Bind time difference` (logical): Determines whether to append a column with the *time difference* since the previous location (units: hours). Default: `TRUE`.
```
same comment as before

bcaneco replied:
same as before

lindesaysh commented on README.md:
```diff
@@ -45,46 +48,45 @@ Move2 location object
     or an error will be returned and processing will stop. Check the logs for the list of column names in the input dataset.
 
--   Any `column` settings can be bugged by a column name with two (and, sometimes, one) periods: for example, 'altitude.col.xyz' could not be recognized. To solve this bug, any column containing a period must be duplicated (instead of renamed) with the new name `altutide/temperature/heading`.
+-   Any `column` settings can be bugged by a column name with two (and, sometimes, one) periods: for example, 'altitude.col.xyz' could not be recognized. To solve this bug, any column containing a period must be duplicated (instead of renamed) with the new name `altitude/temperature/heading`.
```
is this something the user has to do?  how would they go about changing a column name of something coming from moveapps?

bcaneco replied:
Yeah, I'm not sure about that statement either. My understanding is that the code is solving the issue automatically, i.e. it duplicates the column when it's name contains a period.  @callumjclarke, can you confirm please?

lindesaysh reviewed (commented):
i've left a few comments.  not sure how this works exactly!

bcaneco commented:
Thanks Lindesay! I'll go over your comments and do required amendments.

bcaneco committed:
amended text on reference calculation point #5

amended text to reflect reference point for speed, distance and lag calculations

bcaneco committed:
text amended #5
